app/models/orphan.rb

```diff
@@ -96,16 +109,12 @@ def eligible_for_sponsorship?
     Orphan.active.currently_unsponsored.include? self
   end
 
-  def currently_sponsored?
-    sponsorships.all_active.present?
-  end
-
   def current_sponsorship
-    sponsorships.all_active.first if currently_sponsored?
+    sponsorships.all_active.first if sponsored?
```

Comment:
Relying on `sponsorship_status` to accurately represent current state instead of dynamically querying whether any active sponsorships are present. Good idea?